src/autiter.rs

```diff
@@ -53,13 +53,13 @@ pub trait Automaton<P>: Sized {
     }
 
     /// Returns an iterator of non-overlapping matches in `s`.
-    fn find<'a, 's>(
+    fn find<'a, 's, Q: ?Sized + AsRef<[u8]>>(
         &'a self,
-        s: &'s str,
+        s: &'s Q,
     ) -> Matches<'a, 's, P, Self> {
```

Comment:
@jneem To clarify, I guess we'd have to remove `Automaton: Sized` for this to work. Can we do that? (I can't remember why it has a `Sized` bound.)